lib/block-supports/layout.php

```diff
@@ -53,14 +54,29 @@ function gutenberg_get_layout_style( $selector, $layout, $has_block_gap_support
 		$wide_max_width_value = wp_strip_all_tags( explode( ';', $wide_max_width_value )[0] );
 
 		if ( $content_size || $wide_size ) {
-			$style  = "$selector > :where(:not(.alignleft):not(.alignright)) {";
+			$style = "$selector {";
+			// Using important here to override the inline padding that could be potentially
+			// applied using the custom padding control before the layout inheritance is applied.
+			$style .= sprintf(
+				'padding: %s %s %s %s !important',
+				isset( $padding['top'] ) ? $padding['top'] : 0,
+				isset( $padding['right'] ) ? $padding['right'] : 0,
+				isset( $padding['bottom'] ) ? $padding['bottom'] : 0,
+				isset( $padding['left'] ) ? $padding['left'] : 0
```

Comment:
Sorry, I will test this manually, but ran out of time today 😄 

So this is a note for myself more than anything.

Let's say the result is `padding: 10px 0 0 0;`, where only `padding-top: 10px` was defined by the user in the editor.

Is there any problems with right, bottom and left being now `0`? 

What if a CSS style higher up says otherwise, and is consequently overridden?

Reply:
Oh, good point!

> What if a CSS style higher up says otherwise, and is consequently overridden?

It sounds like that might be a similar challenge to what we've been discussing in the PR to add support for `blockGap` at the block level within global styles: https://github.com/WordPress/gutenberg/pull/39870#discussion_r841413547 — TL;DR: how do we reliably get the value that's cascaded down correctly at this point in the execution. E.g. do we need to look up global styles here as well, to get the right value?

Reply:
Actually, wait, I think I might be getting confused here — because this is a different `padding` that comes from the layout object instead?

Reply:
> E.g. do we need to look up global styles here as well, to get the right value?

I think we might, yes. Though ideally, either the block attributes would know about global styles too, or that cascade would be resolved someplace else by the time we get to layout. Is that within the scope of the style engine work?

Reply:
> Is that within the scope of the style engine work?

I'm not _too_ sure, to be honest, it's at least beyond the initial work of refactoring the existing block supports, but I think we might naturally bump into it when we go to consolidate style generation for global styles / theme.json.

Reply:
Just rounding back to this: global/theme.json block styles are overwritten for the reasons @andrewserong [mentions above](https://github.com/WordPress/gutenberg/pull/39871#discussion_r847013686)

<img width="284" alt="Screen Shot 2022-04-12 at 3 18 09 pm" src="https://user-images.githubusercontent.com/6458278/162886028-87b38538-4548-4d9d-908c-74a06072bc9f.png">

> Is that within the scope of the style engine work?

I see the intention is for `!important` to override inline styles. Works as intended, but the inline styles are still there.

Without promising anything 😄 I also wonder if the style engine will later dedupe this sort of stuff as well. Layout crosses over into spacing more and more it seems!

Reply:
Can you clarify why we need to render the padding here (I mean in layout)? Can't we assume the padding block support will just output the same thing (inline or using a style tag) later?

Reply:
I guess that's for the "inherited padding", maybe we can only output these styles if the layout is inherited to avoid duplication and avoid important?